src/EventGridExtension/EventGridTriggerAttributeBindingProvider.cs

```diff
@@ -96,8 +115,8 @@ public Type TriggerValueType
             public async Task<ITriggerData> BindAsync(object value, ValueBindingContext context)
             {
                 EventGridEvent triggerValue = value as EventGridEvent;
-                var bindingData = await _publisher.ExtractBindingData(triggerValue, _parameter.ParameterType);
-                IValueBinder valueBinder = new EventGridValueBinder(_parameter, _publisher.GetArgument(bindingData), _publisher.Recycles);
+                var bindingData = await ExtractBindingData(triggerValue, _parameter.ParameterType);
+                IValueBinder valueBinder = new EventGridValueBinder(_parameter, GetArgument(bindingData));
```

Comment:
Don't have GetArgument(bindingData). Instead, just pass value directly.  If we need to coerce it to the user's parameter type, we can coerce it here.